src/vs/workbench/browser/parts/editor/editorControl.ts

```diff
@@ -48,15 +57,29 @@ export class EditorControl extends Disposable {
 		private groupView: IEditorGroupView,
 		@IWorkbenchLayoutService private readonly layoutService: IWorkbenchLayoutService,
 		@IInstantiationService private readonly instantiationService: IInstantiationService,
-		@IEditorProgressService private readonly editorProgressService: IEditorProgressService
+		@IEditorProgressService private readonly editorProgressService: IEditorProgressService,
+		@IWorkspaceTrustManagementService private readonly workspaceTrustService: IWorkspaceTrustManagementService,
 	) {
 		super();
 	}
 
 	async openEditor(editor: EditorInput, options: EditorOptions | undefined, context: IEditorOpenContext): Promise<IOpenEditorResult> {
+		const requiresTrust = await editor.requiresTrust();
+		const blockedByTrust = requiresTrust && !this.workspaceTrustService.isWorkpaceTrusted();
+
+		if (requiresTrust) {
+			const disposable = this.workspaceTrustService.onDidChangeTrust(() => {
```

Comment:
Not so happy about accumulating a listener each time an editor opens, wouldn't it be better to listen to trust changes from the constructor and then update the active pane accordingly if it is a workspace trust editor?